tests/headers/referer_test.go

```diff
@@ -105,3 +93,57 @@ func TestReferer(t *testing.T) {
 		})
 	}
 }
+
+func TestRefererOrdering(t *testing.T) {
+	// It is not clear from the documentation of http.Request.Referer()
+	// that only the first Referer header is used and any other ones
+	// are ignored. This could potentially lead to security issues if two
+	// HTTP servers are chained that look at different headers.
+	//
+	// The desired behavior would instead be for http.Request.Referer() to
+	// return an error of some sort or an empty string when there are more
```

Comment:
Can you give an example of some error?

Reply:
For `Referer()` and `UserAgent()` they don't currently return an error, only a string, so I thought that to signal that something went wrong we might return an empty string. But I think it would be nicer to add an error return value. The kind of error doesn't really matter as long as it is clear what went wrong. So some examples would be:
```go
errors.New("multiple Referer headers")
errors.New("multiple User-Agent headers")
```

Should this amount of detail really be in the comment though? I think that the specific error message would be up to the person implementing this.

Reply:
I think that here we have to do a risk evaluation: getting the referer wrong could have secuirty implications, getting the UA wrong, not too much.

In any case I don't think we should have too strong opinion on the matter as it is normal practice for cooperating clients to not send duplicate request headers, so for our web framework we'll reject other requests, but for the stdlib I guess it is okay to accept odd requests.

Reply:
I'm gonna go with that the desired behavior should be the behavior we want in our framework. I will therefore change the desired behavior for both tests to be to respond with a `400 Bad Request`.